### PR TITLE
adds point in polygon to Point3.

### LINKED ISSUE #86: Point contained inside a polygon
The method used to check if a point is inside a polygon.
https://stackoverflow.com/questions/217578/how-can-i-determine-whether-a-2d-point-is-within-a-polygon

### PR DESCRIPTION
**Pull request by d3ssy**
### What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

### Description
- Adds point-in-polygon test to Point3.
- Works with rotated polygons and points in 3d.
- Cleaned up some doc strings.

### Related Tickets & Documents
Closes #86 

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [x] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/GShark/Geometry/Point3.cs [hunk 1]
```diff
@@ -590,14 +590,68 @@ public static Point3[] CullDuplicates(IEnumerable<Point3> points, double toleran
...
         }
+
+        /// <summary>
+        /// Tests whether a point is inside, outside, or coincident with a polygon.
+        /// <para>See https://stackoverflow.com/a/16391873</para>
+        /// </summary>
+        /// <param name="point">The point to test.</param>
+        /// <param name="polygon">The polygon to test against.</param>
+        /// <returns>Returns 0 if the point is outside the polygon or 1 if it is inside.</returns>        
+        public int InPolygon(Polygon polygon)
```
Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:610]: What does it returns if it is coincident?

Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:610]: Could we have this operation at the level of the `Curve`
It sounds natural to think we are operating on the curve instead of the point.
What about moving this method to `NurbsBase`?

**Review by sonomirco — COMMENTED** [review]
Thanks, @d3ssy :)

File: src/GShark/Geometry/Point3.cs (hunk 1 above)
Comment (d3ssy) [reply src/GShark/Geometry/Point3.cs:610]: This method is only applicable to Polygon as they are guaranteed to be planar curves. PointInCurve() given an arbitrary 3d curve is another can of worms. 

I placed it in Point3 to match the Point3.InPlane() method.

Comment (sonomirco) [reply src/GShark/Geometry/Point3.cs:610]: Shouldn't every curve be planar to perform this operation?

Comment (d3ssy) [reply src/GShark/Geometry/Point3.cs:609]: Try and it in rhino and see 😄.

Comment (d3ssy) [reply src/GShark/Geometry/Point3.cs:609]: Updated method to test for coincidence edge cases.

Comment (d3ssy) [reply src/GShark/Geometry/Point3.cs:609]: Regardless of planarity, we will need a more computationally intensive method for NURBS that calculates ray intersections, and does closest point for coincidence checks. The logic should be the same. We can limit it to planar curves in the case of NURBS and throw error if it ain't.
